### ISSUE TITLE
Function-based mock response cannot change HTTP status

### ISSUE THREAD

**Issue by JohnAlbin** [nutboltu/storybook-addon-mock#116]
I have a fetch endpoint that various authentication tasks and they all use the same URL; the endpoint returns different data depending on the request body.

In one story, I have a page component that needs to have the client credential request respond with HTTP 200 immediately followed by another endpoint request that should respond with HTTP 500.

Currently, each mocked endpoint can only respond with one static status number.

**Comment by yormy** [comment]
would be nice indeed to have that ability

**Comment by JohnAlbin** [comment]
@yormy This missing feature was a deal breaker for me; I needed to test successful endpoint responses _and_ failures. I ended up writing my own Storybook wrapper around the fetch-mock library.

https://github.com/JohnAlbin/storybook-addon-fetch-mock

**Comment by iliakan** [comment]
@JohnAlbin your code only mocks fetch, but Angular uses XHR, so it's important to note that it's not compatible with Angular apps. Unlike this module =)

**Comment by JohnAlbin** [comment]
> your code only mocks fetch, but Angular uses XHR, so it's important to note that it's not compatible with Angular apps. 

Fair enough. But if an app needs XHR mocking, it's pretty clear `storybook-addon-fetch-mock` only does fetch mocking.

> And yeah, this module is mockingly powerless to do real mocking. Can't set headers etc, that's just pathetic.

Don't be unkind. Just because `storybook-addon-mock` doesn't solve our issues, doesn't mean it can't solve lots of other people's issues.

**Comment by iliakan** [comment]
> Don't be unkind. Just because `storybook-addon-mock` doesn't solve our issues, doesn't mean it can't solve lots of other people's issues.

Well, after I invested time into integrating this module, it just stops working. That's quite unkind from the module.

Imagine you take a car, it goes 10km and stops. Not what you'd expect ;(

**Comment by yormy** [comment]
Added MR for customized status code returns
